### PR TITLE
Detect whether the client is connected to the internet. Only allow client calls if connected.

### LINKED ISSUE #82: Fail gracefully without network access
Currently, trying to `bundle exec jekyll serve` a page not even using jekyll-github-metadata fails, because the jekyll-github-metadata plugin is (unconditionally, afaict) pulled in as a dependency of github-pages.

The failure mode is like this:
```
% bundle exec jekyll serve       
Configuration file: /home/user/qubesos.github.io/_config.yml
Configuration file: /home/user/qubesos.github.io/_config.yml
            Source: /home/user/qubesos.github.io
       Destination: /home/user/qubesos.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
                    done in 9.53 seconds.
 Auto-regeneration: enabled for '/home/user/qubesos.github.io'
Configuration file: /home/user/qubesos.github.io/_config.yml
    Server address: http://127.0.0.1:4000/
  Server running... press ctrl-c to stop.
      Regenerating: 1 file(s) changed at 2016-12-11 08:05:11    GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.
   GitHub Metadata: Failed to open TCP connection to api.github.com:443 (No route to host - connect(2) for "api.github.com" port 443)
   GitHub Metadata: Failed to open TCP connection to api.github.com:443 (No route to host - connect(2) for "api.github.com" port 443)
   GitHub Metadata: Failed to open TCP connection to api.github.com:443 (No route to host - connect(2) for "api.github.com" port 443)
   GitHub Metadata: Failed to open TCP connection to api.github.com:443 (No route to host - connect(2) for "api.github.com" port 443)
   GitHub Metadata: Failed to open TCP connection to api.github.com:443 (No route to host - connect(2) for "api.github.com" port 443)
   GitHub Metadata: Failed to open TCP connection to api.github.com:443 (No route to host - connect(2) for "api.github.com" port 443)
```

See also: https://github.com/github/pages-gem/issues/364

### PR DESCRIPTION
**Pull request by parkr**
Still very WIP, but this fixes #82.

### CODE REVIEW

**Comment by parkr** [comment]
@benbalter What do you think of the overall approach, here?

File: lib/jekyll-github-metadata/client.rb [hunk 1]
```diff
@@ -59,6 +59,8 @@ def method_missing(method_name, *args, &block)
       end
 
       def save_from_errors(default = false)
+        Jekyll::GitHubMetadata.log :warn, "No internet connection." unless internet_connected?
```
Comment (benbalter) [inline lib/jekyll-github-metadata/client.rb]: I'd add something like "GitHub metadata may be missing or incorrect".

Comment (parkr) [inline lib/jekyll-github-metadata/client.rb]: Fixed!

**Comment by benbalter** [comment]
@jekyllbot: merge +minor.

**Pull request merged**